lib/targets/ara.c

```diff
@@ -10,9 +10,9 @@ uint64_t rdcycle64(){
     uint32_t cyclesh2;
     do
     {
-        cyclesh1 = rdcycleh()
-        cyclesh2 = rdcycle()
-        cyclesh3 = rdcycleh()
+        cyclesh1 = rdcycleh();
+        cycles = rdcycle();
+        cyclesh2 = rdcycleh();
```

Comment:
Can you give a short explanation on what was wrong here? We are probably able to use the same methods as in `generic_riscv.c`? Maybe it would make sense to move the common code (to handle the CSRs) to a common file.